app/src/main/java/org/sil/gatherwords/DatabaseAccess.java

```diff
@@ -0,0 +1,139 @@
+package org.sil.gatherwords;
+
+import android.os.AsyncTask;
+
+import org.sil.gatherwords.room.AppDatabase;
+import org.sil.gatherwords.room.Session;
+import org.sil.gatherwords.room.SessionDao;
+import org.sil.gatherwords.room.Word;
+import org.sil.gatherwords.room.WordDao;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Accesses the database in a worker thread.
+ * Prevents blocking the main thread
+ */
+
+public class DatabaseAccess extends AsyncTask<String, Void, List<?>> {
+	private AppDatabase appDatabase;
+	private SessionDao sd;
+	private WordDao wd;
+
+	private Session[] sessions;
+	private Word[] words;
+	private List<String> stringList;
+	private String comparisonString;
+
+	/**
+	 * Construct the task
+	 * @param ad The database instance because this class does not have access to getApplicationContext()
+	 */
+	DatabaseAccess(AppDatabase ad){
+		appDatabase = ad;
+	}
+
+	@Override
+	protected void onPreExecute() {
+		super.onPreExecute();
+		sd = appDatabase.sessionDao();
+		wd = appDatabase.wordDao();
+	}
+
+	/**
+	 * Run all database access in a worker thread, so that the main thread does not get blocked
+	 * @param strings Which function insert, select, update, delete to run
+	 * @return null
+	 */
+	@Override
+	protected List<?> doInBackground(String... strings) {
+		List<?> result = null;
+		appDatabase.beginTransaction();
+		if (strings[0].equals("insert")) {
+			sd.insertSession(sessions);
+			wd.insertWords(words);
+		}
+		if (strings[0].equals("select")) {
+			if (strings[1].equals("session")) {
+				if (stringList == null && comparisonString == null) {
+					result = sd.getAll();
+				} else if (stringList == null && comparisonString != null) {
+					stringList = Arrays.asList("id", "label", "date", "speaker", "recorder", "vernacular", "listLanguages", "location", "gps");
+					result = sd.getWhere(stringList, comparisonString);
+				} else if (stringList != null && comparisonString == null) {
+					result = sd.getAll(stringList);
+				} else {
+					result = sd.getWhere(stringList, comparisonString);
+				}
```

Comment:
The purpose of this set of alternates is not immediately visible.  Are all execution paths actually in use?  If so, please leave a comment what each mean.  Otherwise remove the unnecessary ones.

Reply:
I should add the path for Words, but this also is where inserts will be (until we do it the proper way). I'll pull master in to make that more obvious